File: cf_1060_2/A_notelook.cpp
#include <bits/stdc++.h>
using namespace std;

int min_protections(int n, int k, const string &s) {
    int ans = 0;
    int j = 0;
    bool flag = true;

    for (int i = 0; i < n; i++) {
        if (s[i] == '1' && flag == false) {
            if (i - j > k - 1) {
                ans++;
                j = i;
            } else {
                j = i;
            }
        } else if (s[i] == '1' && flag == true) {
            ans++;
            flag = false;
            j = i;
        }
    }

    return ans;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;
    while (t--) {
        int n, k;
        cin >> n >> k;
        string s;
        cin >> s;  // string input (e.g., "10101")
        cout << min_protections(n, k, s) << "\n";
    }

    return 0;
}

File: cf_1060_2/A_Notelook.py
def min_protections(n, k, s):
    ans=0
    j=0
    flag=True
    for i in range(n):
        if s[i]=='1' and flag==False:
            if i-j>k-1:
                ans+=1
                j=i
            else:
                j=i
        elif s[i]=='1' and flag==True:
            ans+=1
            flag=False
            j=i
            
    return ans
              
         

t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    s = list(input().strip())
    print(min_protections(n, k, s))
